engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -251,7 +255,19 @@ private[spel] class Typer(classLoader: ClassLoader, commonSupertypeFinder: Commo
         case _ => invalid("Invalid ternary operator") // shouldn't happen
       }
       //TODO: what should be here?
-      case e: TypeReference => fixed(Unknown)
+
+      case e: TypeReference => {
+
+        if(!referenceTypeValidating) {
```

Comment:
Try to not use return. Look how we do it above in this class.

Reply:
Do something like this:
```
(referenceTypeValidating, typeDefinitionSet.validateTypeReference(e)) {
case (false, _) => valid(Unknown)
case (true, true) => valid(Unknown)
_ => invalid("Class is not allowed to be passed as TypeReference")
}
```
After that you shouldn't use return :)

Reply:
I think I fixed it. Please take a look again when you find the time if that's what you meant

Reply:
Ok, It looks better now.